### DISCUSSION TITLE
Can I set all error messages to automatically include scope/file/line number information?

### DISCUSSION THREAD

**Discussion by sheldonhull** [pterm/pterm#299] [Q&A] [answered]
I understand I can use PrefixPrinter and use options like WithLineNumber.
However, is it possible to set that behavior as the default for `pterm.Error.Printfln` so I don't need to specify it each time? 

I'm guessing I could just implement my own function, but asking as it seems useful in any errors to include this for my use case.

**Comment by MarvinJWendt** [comment, marked as answer]
Hi @sheldonhull, yes this is very possible! 😄 
Here is the documentation about how to configure the default printers: https://pterm.sh/#/docs/customizing/override-default-printer

For your case:

```go
	pterm.Error = *pterm.Error.WithShowLineNumber() // Configure the printer here when your application starts
    // can also be put into the init() method of your package to call it before everything else runs

	pterm.Error.Printfln("%s", "Hello World")
	pterm.Error.Printf("%s\n", "Hello World")
	pterm.Error.Print("Hello World\n")
	pterm.Error.Println("Hello World")
```

-->

![image](https://user-images.githubusercontent.com/31022056/147391740-b336bd1b-c88e-41cc-99e8-ea1361e99dbf.png)


While writing this, I just noticed that some show the wrong line numbers. This is because the offset for the different functions is different. I'll fix this soon!

**Reply by sheldonhull** [reply]
Thank you @MarvinJWendt. Yes I was a bit confused when I tried with line numbers as I had to offset by 1 to get my file and not pterms file number. 
The verbosity for line number was a lot to include each time, so I like your example, as it helps provide a more concise way to set this as a default in the entry of the program. Thanks for the help!

**Reply by sheldonhull** [reply]
@MarvinJWendt I tried this just now and see it still doing prefix printer as the output, rather than providing the original source file. I'm on `v0.12.46`. Do I still need to use offset?

**Reply by MarvinJWendt** [reply]
> @MarvinJWendt I tried this just now and see it still doing prefix printer as the output, rather than providing the original source file. I'm on `v0.12.46`. Do I still need to use offset?

Sorry, I seem to have missed that reply. I am currently working on this, should be fixed today :) (Hopefully)

**Reply by MarvinJWendt** [reply]
@sheldonhull the PR is ready and will be merged in a couple of minutes :)
https://github.com/pterm/pterm/pull/437

**Reply by sheldonhull** [reply]
I use pterm all over the place. Thanks for your hard work on this. Such a great package! 🥇

**Reply by MarvinJWendt** [reply]
Glad that you like it! Let me know if you ever need help :)
